### PR TITLE
Adding support for new Waffling60 revision

### PR DESCRIPTION
**Pull request by 4pplet**
## Description

Adding QMK support for the latest revision of waffling60. Revision E adds underglow and 6u space support. 

## Types of Changes
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/4pplet/waffling60/rev_e/info.json [hunk 1]
```diff
@@ -0,0 +1,651 @@
...
+    "bootloader": "stm32-dfu",
+    "diode_direction": "COL2ROW",
+    "encoder": {
+        "rotary": [
+            {"pin_a": "A2", "pin_b": "A1", "resolution": 2}
+        ]
+    },
+    "features": {
+        "audio": false,
+        "backlight": false,
```
Comment (dunk2k) [inline keyboards/4pplet/waffling60/rev_e/info.json:14]: not enabled by default
```suggestion
        "audio": false,
        "backlight": false,
```

File: keyboards/4pplet/waffling60/rev_e/keymaps/via/rules.mk [hunk 2]
```diff
@@ -0,0 +1 @@
+VIA_ENABLE = yes
```
Comment (dunk2k) [inline keyboards/4pplet/waffling60/rev_e/keymaps/via/rules.mk:1]: Required for conditional encoder_map_enable code in `keymap.c` to trigger
```suggestion
VIA_ENABLE = yes
ENCODER_MAP_ENABLE = yes
```

File: keyboards/4pplet/waffling60/rev_e/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,            KC_TRNS,              KC_TRNS,              KC_TRNS,               KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS)
+};
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+    [0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [1] = { ENCODER_CCW_CW(KC_BRID, KC_BRIU) },
+    [2] = { ENCODER_CCW_CW(KC_MPRV, KC_MNXT) },
+    [3] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) }
```
Comment (dunk2k) [inline keyboards/4pplet/waffling60/rev_e/keymaps/via/keymap.c:54]: 1. non-functional layers removed
```suggestion
  KC_TRNS,  KC_TRNS,  KC_TRNS,            KC_TRNS,              KC_TRNS,              KC_TRNS,               KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS)
};
#if defined(ENCODER_MAP_ENABLE)
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
    [0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
    [1] = { ENCODER_CCW_CW(KC_BRID, KC_BRIU) }
```

File: keyboards/4pplet/waffling60/rev_e/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+// basic function layer
+[1] = LAYOUT_all(
+  QK_BOOT,  KC_F1,    KC_F2,    KC_F3,    KC_F4,     KC_F5,     KC_F6,     KC_F7,     KC_F8,     KC_F9,      KC_F10,  KC_F11,   KC_F12,  KC_TRNS, KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_UP,    KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS,
+  KC_TRNS,  KC_LEFT,  KC_DOWN,  KC_RIGHT, KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,            KC_TRNS,              KC_TRNS,              KC_TRNS,               KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS)
+};
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (dunk2k) [inline keyboards/4pplet/waffling60/rev_e/keymaps/default/keymap.c:36]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by dunk2k** [comment]
Could you add a `rules.mk` file to `keymaps/default` directory with contents:
```
ENCODER_MAP_ENABLE = yes

```

File: keyboards/4pplet/waffling60/rev_e/rev_e.h [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#define CAPS_LOCK_ENABLE true
+#define CAPS_LOCK_COLOR HSV_GREEN
+
+#include "quantum.h"
```
Comment (fauxpark) [inline keyboards/4pplet/waffling60/rev_e/rev_e.h:22]:
```suggestion
#include "quantum.h"

#define CAPS_LOCK_ENABLE true
#define CAPS_LOCK_COLOR HSV_GREEN
```

File: keyboards/4pplet/waffling60/rev_e/rules.mk [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# Build Options
+#   change yes to no to disable
+#
+KEY_LOCK_ENABLE = yes
+
+# Wildcard to allow APM32 MCU
+DFU_SUFFIX_ARGS = -p FFFF -v FFFF
+
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
```
Comment (fauxpark) [inline keyboards/4pplet/waffling60/rev_e/rules.mk:10]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by 4pplet** [comment]
> @4pplet Could you add a `rules.mk` file to `keymaps/default` directory with contents:
> 
> ```
> ENCODER_MAP_ENABLE = yes
> ```

Added now! :)

File: keyboards/4pplet/waffling60/rev_e/info.json (hunk 1 above)
Comment (dunk2k) [inline keyboards/4pplet/waffling60/rev_e/info.json:14]: not enabled by default
```suggestion
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/4pplet/waffling60/rev_e/rules.mk [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# Build Options
+#   change yes to no to disable
+#
+KEY_LOCK_ENABLE = yes
```
Comment (dunk2k) [inline keyboards/4pplet/waffling60/rev_e/rules.mk:4]: migrate to `info.json`
```suggestion
```

File: keyboards/4pplet/waffling60/rev_e/info.json [hunk 8]
```diff
@@ -0,0 +1,649 @@
...
+    "encoder": {
+        "rotary": [
+            {"pin_a": "A2", "pin_b": "A1", "resolution": 2}
+        ]
+    },
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
```
Comment (dunk2k) [inline keyboards/4pplet/waffling60/rev_e/info.json:16]: migrated from `rules.mk`
```suggestion
        "extrakey": true,
        "key_lock": true,
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by 4pplet** [comment]
Any update on this? This should be ready for merge?

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
> Any update on this? This should be ready for merge?

Sorry, the last couple of weeks are/were breaking changes time, so ... busy and merge blocked.

**Review by fauxpark — CHANGES_REQUESTED** [review]
This PR should go to `develop`, as it modifies existing keyboard files.

File: keyboards/4pplet/waffling60/rev_e/rules.mk [hunk 9]
```diff
@@ -0,0 +1,6 @@
+# Build Options
+#   change yes to no to disable
+#
+
```
Comment (fauxpark) [inline keyboards/4pplet/waffling60/rev_e/rules.mk:4]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by dunk2k** [comment]
@4pplet 

Reading `readme.md` says there is 6u (space bar) support but not seeing anything here that show it. Would you be able to provide a `matrix_diagram.md`?

**Comment by drashna** [comment]
> @4pplet
> 
> Reading `readme.md` says there is 6u (space bar) support but not seeing anything here that shows it. Would you be able to provide a `matrix_diagram.md`?

Yeah, it's a bit odd to call it out in the readme, but not actually support it in the config. 

One or the other should go, but it sounds like ideally, the config should be updated.

**Comment by 4pplet** [comment]
For sure, I'll add layouts in info.json for: 6u ISO, 6u ISO split backspace, 6u ANSI, 6u ANSI split backspace before merge. That should be fine?

**Comment by dunk2k** [comment]
> For sure, I'll add layouts in info.json for: 6u ISO, 6u ISO split backspace, 6u ANSI, 6u ANSI split backspace before merge. That should be fine? And a matrix diagram

Yes please, thank you.

**Review by drashna — APPROVED** [review]

**Comment by 4pplet** [comment]
This should be ready for merge?

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**